### DISCUSSION TITLE
Replacing Adafruit_GFX with Arduino_GFX?

### DISCUSSION THREAD

**Discussion by VerusLogic** [moononournation/Arduino_GFX#237] [Q&A]
I have an older project that we used Adafruit_GFX.  So there are many calls in my code that use 

tft.foo()

The Arduino_GFX uses 
gfx->foo()

Is there a way to define the tft class to use this way rather than a pointer notation?

The alternative is to replace all "tft." in my code with "gfx->" which shouldn't be a problem.  I see Arduino_TFT.h which looks like the place that does this, but I can't make it work this way.

Are there any examples for how to setup a class like this?

Arduino_TFT tft

so that the code can be reused?

tft.begin();

Thanks,

Mike.

**Comment by moononournation** [comment]
you can replace `Arduino_GFX *gfx = new Arduino_ILI9341(` to `Arduino_GFX tft = Arduino_ILI9341(`

**Reply by VerusLogic** [reply]
Thanks.  I thought I tried that.  

`Arduino_GFX tft = Arduino_ILI9488(bus, DF_GFX_RST, 3 /* rotation */, true /* IPS */);`

When` I tried it again I got this issue:

`object of abstract class type "Arduino_GFX" is not allowed:C/C++(322)
AC_display.cpp(45, 13): function "Arduino_GFX::begin" is a pure virtual function
AC_display.cpp(45, 13): function "Arduino_GFX::writePixelPreclipped" is a pure virtual function
cannot declare variable 'tft' to be of abstract type 'Arduino_GFX'`

I'm not that familiar with abstract class.  I have to educate myself on that.

Any additional advice?

Thanks,

Mike

**Reply by moononournation** [reply]
then try `Arduino_ILI9488 tft = Arduino_ILI9488(bus, DF_GFX_RST, 3 /* rotation */, true /* IPS */);`

**Comment by VerusLogic** [comment]
I appreciate the quick response.  I tried `Arduino_ILI9488 tft = Arduino_ILI9488(bus, DF_GFX_RST, 3 /* rotation */, true /* IPS */);`

But I got a similar error:

`Arduino_ILI9488 tft
More display class: https://github.com/moononournation/Arduino_GFX/wiki/Display-Class

conflicting declaration 'Arduino_ILI9488 tft'`

Your library is fantastic and your work is much appreciated.  I guess I need to convert my old code to the pointer to the class and continue.

I am curious what the Arduino_TFT.h file is for.  This seems close to what I was hoping for.  

Is there a way to create a class that wrappers the library calls to use the class like the Adafruit_GFX library.

Thanks,

Mike.

**Reply by moononournation** [reply]
I have tested `Arduino_ILI9488 tft = Arduino_ILI9488(bus, DF_GFX_RST, 3 /* rotation */, true /* IPS */);`, it should compile without error.

**Comment by VerusLogic** [comment]
Interesting. I will give it another try in a smaller project.

Thanks.


On Dec 18, 2022, at 7:55 PM, moononournation ***@***.***> wrote:

﻿

I have tested Arduino_ILI9488 tft = Arduino_ILI9488(bus, DF_GFX_RST, 3 /* rotation */, true /* IPS */);, it should compile without error.

**Comment by VerusLogic** [comment]
That worked!  Thanks so much for your patience and your wonderful library.
